File: core/src/main/java/com/metricstream/jdbc/LongString.java
/*
 * Copyright © 2020-2021, MetricStream, Inc. All rights reserved.
 */
package com.metricstream.jdbc;

import java.io.Reader;
import java.io.StringReader;


public class LongString {
    
    private final String data;
    
    public LongString(String data) {
        this.data = data;
    }
        
    public Reader getReader() {
        return new StringReader(data);
    }
    
    public String getData() {
        return data;
    }
    
    @Override
    public boolean equals(Object object) {
        if (object instanceof LongString) {
            return data.equals(((LongString) object).getData());
        }
        return false;
    }
}

File: core/src/main/java/com/metricstream/jdbc/LongString.kt
/*
 * Copyright © 2020-2022, MetricStream, Inc. All rights reserved.
 */
package com.metricstream.jdbc

import java.io.Reader
import java.io.StringReader

class LongString(val data: String) {
    val reader: Reader
        get() = StringReader(data)

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as LongString

        if (data != other.data) return false

        return true
    }

    override fun hashCode(): Int {
        return data.hashCode()
    }
}
